### PR TITLE
config: Fix confusing errors

### PR DESCRIPTION
**Pull request by hsanjuan**
The JSON parsing of the config could error, but we skipped error checking and
use Validate() at the end. This caused that maybe some JSON parsing errors
where logged but the final error when validating the configuration came from
somewhere different, creating very confusing error messages for the user.

This changes this, along with removing hardcoded section lists. This also
removes a Sharder component section because, AFAIK, it was a left over
from the sharding branch and in the end there is no separate sharding
component that needs a config.

License: MIT
Signed-off-by: Hector Sanjuan <code@hector.link>

### CODE REVIEW

**Comment by hsanjuan** [comment]
Hmm I need to fix a small quirk with this

File: config/config.go [hunk 1]
```diff
@@ -165,7 +174,29 @@ type jsonConfig struct {
 	Monitor    jsonSection      `json:"monitor,omitempty"`
 	Allocator  jsonSection      `json:"allocator,omitempty"`
 	Informer   jsonSection      `json:"informer,omitempty"`
-	Sharder    jsonSection      `json:"sharder,omitempty"`
+}
+
+func (jcfg *jsonConfig) getSection(i SectionType) jsonSection {
+	switch i {
+	case Consensus:
```
Comment (kishansagathiya) [inline config/config.go]: no `Cluster` section here?

File: config/util.go [hunk 2]
```diff
@@ -96,6 +96,11 @@ type DurationOpt struct {
 // ParseDurations takes a time.Duration src and saves it to the given dst.
 func ParseDurations(component string, args ...*DurationOpt) error {
 	for _, arg := range args {
+		if arg.Duration == "" {
```
Comment (kishansagathiya) [inline config/util.go]: Any duration field if not passed will take the default value, right?

Comment (hsanjuan) [reply config/util.go]: yes, since all configs are defaulted before being parsed.

File: config/config.go (hunk 1 above)
Comment (hsanjuan) [reply config/config.go]: Fixed (it's not used and I forgot).

Comment (hsanjuan) [reply config/config.go]: No, actually there is not a section for Clusterbecause it's a *json.RawMessage and it's handled differently. So this was correct.

**Comment by hsanjuan** [comment]
I'll merge this since I'm going to release.

**Pull request merged**